### PR TITLE
Migrate to ClassGraph (formerly FastClasspathScanner) 4.0

### PR DESCRIPTION
**Pull request by PascalSchumacher**
Not ready to merge (final 4.0.0 release not yet available).

Followed migration guide at: https://github.com/classgraph/classgraph/wiki/Porting-FastClasspathScanner-code-to-ClassGraph

Just to provide some feedback for classgraph/classgraph#215.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### CODE REVIEW

File: documentation/src/test/java/org/junit/api/tools/ApiReportGenerator.java [hunk 1]
```diff
@@ -62,24 +63,23 @@ ApiReport generateReport(String... packages) {
 		final String EOL = System.lineSeparator();
 
 		// Scan packages
-		ScanResult scanResult = new FastClasspathScanner(packages).scan();
+		ScanResult scanResult = new ClassGraph().whitelistPackages(packages).enableAnnotationInfo().scan();
 
 		// Collect names
-		List<String> names = new ArrayList<>();
-		names.addAll(scanResult.getNamesOfClassesWithAnnotation(API.class));
-		names.addAll(scanResult.getNamesOfAnnotationsWithMetaAnnotation(API.class));
```
Comment (PascalSchumacher) [inline documentation/src/test/java/org/junit/api/tools/ApiReportGenerator.java]: I'm not sure if this change is correct (see: https://github.com/classgraph/classgraph/issues/215#issuecomment-408674710)

Comment (PascalSchumacher) [reply documentation/src/test/java/org/junit/api/tools/ApiReportGenerator.java]: `Annotation APIs now handle meta-annotations` (source: https://github.com/classgraph/classgraph/wiki/Porting-FastClasspathScanner-code-to-ClassGraph), so this change is fine.

**Comment by PascalSchumacher** [comment]
Pull request updated as ClassGraph 4.0.0 has been released.

**Comment by sbrannen** [comment]
Thanks for the PR!

Tentatively slated for 5.3 RC1 for consideration.

**Comment by sormuras** [comment]
Thanks @PascalSchumacher

**Pull request merged**